sdk/storage/Azure.Storage.DataMovement/src/TransferProgressTracker.cs

```diff
@@ -84,12 +82,15 @@ public void IncrementQueuedFiles()
         /// <param name="bytesTransferred"></param>
         public void IncrementBytesTransferred(long bytesTransferred)
         {
-            if (_options.TrackBytesTransferred)
+            if (_options != default)
             {
-                lock (_bytesTransferredLock)
+                if (_options.TrackBytesTransferred)
```

Comment:
nit: Could you combine these ifs into just `if (_options?.TrackBytesTransferred)`? I assume `default` (null) evaluates to false.

Reply:
That's what I wanted to do initially but there's a compiler error since bool? cannot be converted to bool?

I can change it to
`if ((bool)(_options?.TrackBytesTransferred))` weirdly enough.